src/plots/cartesian/axes.js

```diff
@@ -611,31 +613,50 @@ axes.calcTicks = function calcTicks(ax) {
     generateTicks();
 
     if(ax.rangebreaks) {
-        var nTicksBefore = tickVals.length;
+        // replace ticks inside breaks that would get a tick
+        if(ax.tickmode === 'auto') {
+            for(var t = 0; t < tickVals.length; t++) {
+                var value = tickVals[t].value;
+                if(ax.maskBreaks(value) === BADNUM) {
+                    // find which break we are in
+                    for(var k = 0; k < ax._rangebreaks.length; k++) {
+                        var brk = ax._rangebreaks[k];
+                        if(value >= brk.min && value < brk.max) {
+                            // replace with break end
+                            tickVals[t] = {
+                                minor: false,
+                                value: brk.max
+                            };
+
+                            break;
+                        }
+                    }
+                }
+            }
+        }
 
-        // remove ticks falling inside rangebreaks
-        tickVals = tickVals.filter(function(d) {
-            return ax.maskBreaks(d.value) !== BADNUM;
+        // sort
+        tickVals = tickVals.sort(function(a, b) {
```

Comment:
Why do we need to sort? We didn't before, did we?

Reply:
Good eye. It was needed with previous commits.
Now it is removed by https://github.com/plotly/plotly.js/pull/4677/commits/ea4d8a2bc5160577132212819b1fb554bf006a67.